test/compilable/test18199.d

```diff
@@ -0,0 +1,19 @@
+// REQUIRED_ARGS:
+// PERMUTE_ARGS:
+// https://issues.dlang.org/show_bug.cgi?id=18199
+
+struct MyStruct {
+    int function() f;
+}
+
+// RHS is struct initializer
+static MyStruct test1 = {
+    { return 1 + 1; }
+};
+
+alias MyFun = int function();
+
+// RHS is function literal
+static MyFun test2 = {
+    return 1 + 1;
+};
```

Comment:
Missing test case for https://issues.dlang.org/show_bug.cgi?id=18199#c2, and comment 5

Reply:
added tests covering bug report, thank you